src/scanpy/tools/_utils.py

```diff
@@ -105,28 +105,33 @@ def preprocess_with_pca(adata, n_pcs: int | None = None, random_state=0):
 
 
 def get_init_pos_from_paga(
-    adata, adjacency=None, random_state=0, neighbors_key=None, obsp=None
+    adata: AnnData,
+    adjacency: spmatrix | None = None,
```

Comment:
Was the `spmatrix` here intentional?  I could see an argument for it because this function doesn't use it (so this type is permissive) but also, it fetches it out of `obsp` if it's not present suggesting that it has an "expected" type

Reply:
I went by what’s actually done with the type, and this one just gets passed to `networkx`, which says it can handle all sparse matrix classes.